moquette-0.17/broker/src/main/java/io/moquette/broker/Server.java

```diff
@@ -181,52 +196,262 @@ public void startServer(IConfig config, List<? extends InterceptHandler> handler
+        final Thread telCollector = new Thread(() -> collectAndSendTelemetryData(config));
+        telCollector.start();
+    }
+
+    private void collectAndSendTelemetryData(IConfig config) {
+        final String uuid = checkOrCreateUUID(config);
+
+        final String telemetryDoc = collectTelemetryData(uuid);
+
+        try {
+            sendTelemetryData(telemetryDoc);
+        } catch (IOException e) {
+            LOG.info("Can't reach the telemetry collector");
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Original exception", e);
+            }
+        }
+    }
+
+    private String checkOrCreateUUID(IConfig config) {
+        final String storagePath = config.getProperty(IConfig.DATA_PATH_PROPERTY_NAME, "");
+        final Path uuidFilePath = Paths.get(storagePath, ".moquette_uuid");
+        if (Files.exists(uuidFilePath)) {
+            try {
+                return new String(Files.readAllBytes(uuidFilePath), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                LOG.error("Problem accessing file path: {}", uuidFilePath, e);
+            }
+        }
+        final UUID uuid = UUID.randomUUID();
+        final FileWriter f;
+        try {
+            f = new FileWriter(uuidFilePath.toFile(), false);
+            f.write(uuid.toString());
+            f.close();
+        } catch (IOException e) {
+            LOG.error("Problem writing new UUID to file path: {}", uuidFilePath, e);
+        }
+
+        return uuid.toString();
+    }
+
+    /**
+     * @return a json string with the content of max mem, jvm version and similar telemetry data.
+     * @param uuid*/
+    private String collectTelemetryData(String uuid) {
+        final String remoteIp = "uncollected";
+        final String os = System.getProperty("os.name");
+        final String cpuArch = System.getProperty("os.arch");
+        final String jvmVersion = System.getProperty("java.specification.version");
+        final String jvmVendor = System.getProperty("java.vendor");
+        final long maxMemory = Runtime.getRuntime().maxMemory();
+        final String maxHeap = maxMemory == Long.MAX_VALUE ? "undefined" : Long.toString(maxMemory);
+
+        return String.format(
+            "{\"os\": \"%s\", " +
+                "\"cpu_arch\": \"%s\", " +
+                "\"jvm_version\": \"%s\", " +
+                "\"jvm_vendor\": \"%s\", " +
+                "\"broker_version\": \"%s\", " +
+                "\"standalone\": %s," +
+                "\"max_heap\": \"%s\", " +
+                "\"remote_ip\": \"%s\", " +
+                "\"uuid\": \"%s\"}",
+            os, cpuArch, jvmVersion, jvmVendor, MOQUETTE_VERSION, this.standalone, maxHeap, remoteIp, uuid);
+    }
+
+    private String retrievePublicIP() {
+        try {
+            URL url = new URL("http://whatismyip.akamai.com");
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected a URL in your code that uses an unencrypted HTTP protocol connection. This might expose sensitive data to third parties. To prevent this, we recommend that you construct all URLs using the secure HTTPS protocol.

[Learn more](https://owasp.org/www-community/vulnerabilities/Insecure_Transport)